jstest/common/utils.py

```diff
@@ -28,13 +29,23 @@ class TimeoutException(Exception):
     pass
 
 
-def execute(cwd, cmd, args=None, quiet=False, strict=True):
+def exec_builtin(cwd, cmd, args, env):
     '''
-    Run the given command.
+    Execute the built-in command.
     '''
-    if args is None:
-        args = []
+    from jstest.common import builtins
+
+    builtin_func = builtins.get(cmd)
+    # FIXME: handle stdout and exitcode values.
+    builtin_func(cwd=cwd, args=args, env=env)
+
+    return '', 0
```

Comment:
This is just a helper function, that only used by the `executor`. There are `exec_shell` and `exec_builtin` functions. The `exec_shell` executes `shell` commands and returns with the output and the exit-code. The `exec_builtin` executes Python defined functions and returns with the same schema like the `exec_shell`. The return value and exit codes are not handled yet, so these are dummy values (that is why there is a FIXME label in the code).

The **import** in the `exec_builtin` function is necessary, since `common.builtins` module imports the `common.utils` module, and the `common.utils` imports the `common.builtins` module. If the import would be at the top of the file, there will be a circular reference.

Reply:
As I said, I understand the purpose, but fine. Nevertheless,  the circular reference is very bad here. Could you please also put a FIXME here to handle this issue later?

Reply:
I've added a FIXME comment for that.